DigitalLearningSolutions.Data.Migrations/202203161448_RestructureAdminAndDelegateTables.cs

```diff
@@ -0,0 +1,189 @@
+                .FromTable("AdminAccounts").ForeignColumn("CategoryID").ToTable("CourseCategories")
+                .PrimaryColumn("CourseCategoryID");
+            Alter.Table("AdminAccounts").AddColumn("EmailVerified").AsDateTime().Nullable();
+
+            Rename.Column("UserAdmin").OnTable("AdminAccounts").To("IsSuperAdmin");
+            Rename.Column("CentreAdmin").OnTable("AdminAccounts").To("IsCentreAdmin");
+            Rename.Column("SummaryReports").OnTable("AdminAccounts").To("IsReportsViewer");
+            Rename.Column("ContentManager").OnTable("AdminAccounts").To("IsContentManager");
+            Rename.Column("ContentCreator").OnTable("AdminAccounts").To("IsContentCreator");
+            Rename.Column("Supervisor").OnTable("AdminAccounts").To("IsSupervisor");
+            Rename.Column("Trainer").OnTable("AdminAccounts").To("IsTrainer");
+            Rename.Column("NominatedSupervisor").OnTable("AdminAccounts").To("IsNominatedSupervisor");
+
+            Rename.Column("Login").OnTable("AdminAccounts").To("Login_deprecated");
+            Rename.Column("Password").OnTable("AdminAccounts").To("Password_deprecated");
+            Rename.Column("ConfigAdmin").OnTable("AdminAccounts").To("ConfigAdmin_deprecated");
+            Rename.Column("Forename").OnTable("AdminAccounts").To("Forename_deprecated");
+            Rename.Column("Surname").OnTable("AdminAccounts").To("Surname_deprecated");
+            Rename.Column("Approved").OnTable("AdminAccounts").To("Approved_deprecated");
+            Rename.Column("PasswordReminder").OnTable("AdminAccounts").To("PasswordReminder_deprecated");
+            Rename.Column("PasswordReminderHash").OnTable("AdminAccounts").To("PasswordReminderHash_deprecated");
+            Rename.Column("PasswordReminderDate").OnTable("AdminAccounts").To("PasswordReminderDate_deprecated");
+            Rename.Column("EITSProfile").OnTable("AdminAccounts").To("EITSProfile_deprecated");
+            Rename.Column("TCAgreed").OnTable("AdminAccounts").To("TCAgreed_deprecated");
+            Rename.Column("FailedLoginCount").OnTable("AdminAccounts").To("FailedLoginCount_deprecated");
+            Rename.Column("ProfileImage").OnTable("AdminAccounts").To("ProfileImage_deprecated");
+            Rename.Column("SkypeHandle").OnTable("AdminAccounts").To("SkypeHandle_deprecated");
+            Rename.Column("PublicSkypeLink").OnTable("AdminAccounts").To("PublicSkypeLink_deprecated");
+            Rename.Column("ResetPasswordID").OnTable("AdminAccounts").To("ResetPasswordID_deprecated");
+
+            Rename.Table("Candidates").To("DelegateAccounts");
+            Alter.Table("DelegateAccounts").AddColumn("UserID").AsInt32().Nullable().ForeignKey("Users", "ID");
+            Alter.Column("DateRegistered").OnTable("DelegateAccounts").AsDateTime().Nullable();
+
+            Delete.Index("IX_Candidates_CandidateNumber").OnTable("DelegateAccounts");

```

Comment:
We are deleting 7 indices here, but leaving 3. What is the reasoning behind the choices?

Reply:
Okay, that makes sense. Looking at the test DB, candidates has a total of 10 indices. It sounds to me like each of these should be modified in some way. Either we drop them if their relevant details are going to be on the Users table, or we drop them and then recreate them again (to correct their table name in their names).

But it looks like 3 have been left untouched (plus an additional that is now untouched with the latest commit):

- IX_Candidates_CentreID_DateRegistered (see comment below)
- IX_Candidates_Active_CentreID_LastName (should probably be deleted)
- IX_Candidates_CentreID_EmailAddress (should probably be renamed)
- IX_Candidates_CentreIDAliasID (should probably be deleted)

Reply:
I've made sure all the foreign keys and indexed on these tables have been renamed correctly. We won't rename keys and indexes on other tables that reference these tables, but it's easy enough to change them here. Those based on deprecated columns have been removed, and the rest have been updated to reflect the new table names.